Rewrite sbin/sync-files.sh

freeformz opened the pull request:
### Relace s3cmd with the aws cli.

I've been having problems with s3cmd and I couldn't figure them out.
I've started to use the aws cli in other things, so trying to keep it
standard.

### Parallel shasums

It's been taking longer and longer for them, so kick them all off
and gather up the results via a fifo. This will max out your CPUs, but
that's a good thing.

### Switch to zsh

fifo/signal handling in bash is brain dead and wouldn't work reliably.
Works fine in zsh.

### Add Ability to ignore files

There are times where there is other work in progress that has added
files to the s3 bucket and I don't want to delete them from the bucket.
This option allows the user to ignore those files.

Commit:
Rewrite sbin/sync-files.sh

Relace s3cmd with the aws cli.

I've been having problems with s3cmd and I couldn't figure them out.
I've started to use the aws cli in other things, so trying to keep it
standard.

Parallel shasums

It's been taking longer and longer for them, so kick them all off
and gather up the results via a fifo. This will max out your CPUs, but
that's a good thing.

Switch to zsh

fifo/signal handling in bash is brain dead and wouldn't work reliably.
Works fine in zsh.

Add Ability to ignore files

There are times where there is other work in progress that has added
files to the s3 bucket and I don't want to delete them from the bucket.
This option allows the user to ignore those files.

jmorrell commented on sbin/sync-files.sh:
```diff
@@ -28,52 +30,95 @@ cd "${td}"
-  s="$(< "${jf}" jq -r '."'${f}'".SHA')"
-  if [ ${#s} -eq 40 ]; then
-    sf="$(shasum "${f}" | cut -d \  -f 1)"
+  local sk="$(< "${jf}" jq -r '."'${f}'".SHA' 2>&1)"
```
I find bash kind of cryptic at the best of times, and that combined with the short filenames like `sk`, `jf`, `u` and `sf` here, make it difficult for me to follow the logic

freeformz replied:
sk == sha known
jf == json file
sf == sha file
u == url

freeformz replied:
But yeah. I can rename them.

jmorrell replied:
none of that was obvious to me 😅

Commit:
Expand short names / exclude ignored files from syncs

Expand short names based on feedback.

I also added the ignored names from both download and upload syncs.
This seems to make the most sense. I had two new files from another
branch that I didn't want to upload yet, but ended up doing so because
they weren't excluded at first.
